File: frontend/src/lib/dispatch/riderDisplay.ts
const MOTORCYCLE_COLOR_HEX: Record<string, string> = {
  rojo: '#DC2626',
  azul: '#2563EB',
  negro: '#111827',
  blanco: '#E5E7EB',
  gris: '#6B7280',
  verde: '#16A34A',
  amarillo: '#EAB308',
  naranja: '#EA580C',
  plata: '#94A3B8',
  cafe: '#92400E',
  café: '#92400E',
  morado: '#7C3AED',
};

export function motorcycleColorHex(value: string): string {
  const trimmed = value.trim();
  if (/^#[0-9a-fA-F]{6}$/.test(trimmed)) return trimmed.toUpperCase();
  return MOTORCYCLE_COLOR_HEX[trimmed.toLowerCase()] ?? '#2563EB';
}

export function vehicleTypeLabel(type: string): string {
  return type === 'moto' ? 'Moto' : type;
}

export function riderPhoneDigits(phone: string): string {
  return phone.replace(/\D/g, '');
}

export function riderTelHref(phone: string): string | null {
  const trimmed = phone.trim();
  const digits = riderPhoneDigits(trimmed);
  if (digits.length < 8) return null;
  return trimmed.startsWith('+') ? `tel:${trimmed.replace(/\s/g, '')}` : `tel:+${digits}`;
}

export function riderWhatsAppHref(
  phone: string,
  firstName: string,
  shortId: string,
): string | null {
  const digits = riderPhoneDigits(phone);
  if (digits.length < 8) return null;
  const text = encodeURIComponent(
    `Hola ${firstName}, te escribo por la entrega ${shortId}.`,
  );
  return `https://wa.me/${digits}?text=${text}`;
}

File: apps/rider/lib/rider_display.dart
import 'config.dart';

const _namedMotorcycleColors = <String, String>{
  'rojo': '#DC2626',
  'azul': '#1D4ED8',
  'negro': '#111827',
  'blanco': '#F8FAFC',
  'gris': '#64748B',
  'verde': '#15803D',
  'amarillo': '#CA8A04',
  'naranja': '#EA580C',
  'plata': '#94A3B8',
  'cafe': '#92400E',
  'café': '#92400E',
  'morado': '#7C3AED',
};

final _hexColor = RegExp(r'^#[0-9a-fA-F]{6}$');

String motorcycleColorHex(String value, {String fallback = '#2563EB'}) {
  final trimmed = value.trim();
  if (_hexColor.hasMatch(trimmed)) {
    return trimmed.toUpperCase();
  }
  return _namedMotorcycleColors[trimmed.toLowerCase()] ?? fallback;
}

String? riderPhotoUrl(String? path) {
  if (path == null) return null;
  final trimmed = path.trim();
  if (trimmed.isEmpty) return null;
  if (trimmed.startsWith('http://') ||
      trimmed.startsWith('https://') ||
      trimmed.startsWith('blob:') ||
      trimmed.startsWith('data:')) {
    return trimmed;
  }
  final base = AppConfig.supabaseUrl.replaceAll(RegExp(r'/+$'), '');
  if (base.isEmpty) return null;
  final normalized = trimmed.replaceFirst(RegExp(r'^/+'), '');
  return '$base/storage/v1/object/public/assets/$normalized';
}

String vehicleTypeLabel([String type = 'moto']) {
  return type == 'moto' ? 'Moto' : type;
}
